convex/lib/validators.ts

```diff
@@ -0,0 +1,37 @@
+import { Validator, v } from 'convex/values'
+import { Doc, TableNames } from '../_generated/dataModel'
+import schema from '../schema'
+
+export const betterV = {
+  id: <T extends TableNames>(tableName: T) => {
+    return v.id(tableName)
+  },
+  doc: <T extends TableNames>(tableName: T): Validator<Doc<T>, false, any> => {
+    const validator = v.object(
+      (schema.tables[tableName] as any).documentSchema
+    ) as Validator<any, any, any>
+    if (validator?.kind !== 'object') {
```

Comment:
I'm realizing now that this isn't right -- it should account for a union of objects as well as any (and probably record, once that exists)